bundles/org.openhab.persistence.dynamodb/src/test/java/org/openhab/persistence/dynamodb/internal/PagingIntegrationTest.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.persistence.dynamodb.internal;
+
+import static org.junit.Assert.*;
+
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.persistence.FilterCriteria;
+import org.openhab.core.persistence.FilterCriteria.Operator;
+import org.openhab.core.persistence.FilterCriteria.Ordering;
+import org.openhab.core.persistence.HistoricItem;
+
+/**
+ *
+ * @author Sami Salonen - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class PagingIntegrationTest extends BaseIntegrationTest {
+
+    private static final String NAME = "number";
+    private static final int STATE_COUNT = 10;
+
+    private static @Nullable ZonedDateTime storeStart;
+
+    @BeforeClass
+    public static void checkService() throws InterruptedException {
+        String msg = "DynamoDB integration tests will be skipped. Did you specify AWS credentials for testing? "
+                + "See BaseIntegrationTest for more details";
+        if (service == null) {
+            System.out.println(msg);
```

Comment:
Is this a debug message left?

Reply:
This is intentional to make the message visible even though logging would not be configured (as is the case typically when you run the tests on eclipse)

I can remove it if that is preferred?

Reply:
Should imho be ok as it is just for the tests.